lib/computer.rb

```diff
@@ -8,29 +8,33 @@ class Computer
   def initialize(grid, ships_list)
     @grid = grid
     @ships_list = ships_list
-    @create_ship = CreateShip.new
   end
 
-  def place_ship
-    while @ships_list.ships.size > 0
-      random_ship = select_random_ship
-      position = valid_position(select_random_position, random_ship.length)
-      random_ship.register_position(position[:x], position[:y], position[:orientation])
-      @grid.add_ship(random_ship)
-    end
+  def ship_placement
+    random_ship = generate_ship
```

Comment:
Why no directly call `select_random_ship` here? I'm not sure that `generate_ship` is really needed